### PR TITLE
Add header to outgoing calls

### PR DESCRIPTION
**Pull request by rushtong**
This PR adds a predefined custom header and value to outgoing calls.
Required for https://broadinstitute.atlassian.net/browse/GAWB-799

The following PRs are related:
https://github.com/broadinstitute/firecloud-develop/pull/134
https://github.com/broadinstitute/firecloud-develop/pull/135
https://github.com/broadinstitute/thurloe/pull/35
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  - Acceptance criteria exists and is met
  - Note any changes to implementation from the description
  - Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  - Authentication
  - Authorization
  - Encryption
  - Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- Review cycle:
  - LR reviews
  - Rest of team may comment on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH
  - Submitter updates documentation as needed
  - Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by davidangb** [comment]
I think this needs .travis.yml updates to handle the new `FIRE_CLOUD_ID` env var.

**Comment by davidangb** [comment]
The passthrough at UserService line 164 needs the header, too, yes? (https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/scala/org/broadinstitute/dsde/firecloud/service/UserService.scala#L164) Or does that one feel safe, since we are forcefully injecting the current user's id?

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala [hunk 1]
```diff
@@ -93,7 +93,7 @@ trait OAuthService extends HttpService with PerRequestCreator with FireCloudDire
 
             // we can't use the standard externalHttpPerRequest here, because the requestContext doesn't have the
             // access token yet; we have to add the token manually
-            val pipeline = addCredentials(OAuth2BearerToken(accessToken)) ~> sendReceive
+            val pipeline = addFireCloudCredentials ~> addCredentials(OAuth2BearerToken(accessToken)) ~> sendReceive
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala]: I believe this is only used to talk to rawls, so the FC credentials are not needed here.

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/FireCloudRequestBuilding.scala [hunk 2]
```diff
@@ -41,6 +44,8 @@ trait FireCloudRequestBuilding extends spray.httpx.RequestBuilding {
   }
 
   // with great power comes great responsibility!
-  def addAdminCredentials = addCredentials(OAuth2BearerToken(HttpGoogleServicesDAO.getAdminUserAccessToken))
+  def addAdminCredentials = addCredentials(OAuth2BearerToken(HttpGoogleServicesDAO.getAdminUserAccessToken)) ~> addFireCloudCredentials
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/FireCloudRequestBuilding.scala]: I believe that addAdminCredentials is used twice in the orch codebase, both in ProfileClient.

First usage talks to Rawls. This usage needs the admin creds, but does NOT need the Thurloe shared secret.
Second usage talks to Thurloe. This usage needs the shared secret, but NOT the admin creds.

I suggest leaving addAdminCredentials as-is (i.e. do not include addFireCloudCredentials), and then change the second usage - the one that talks to Thurloe - to remove addAdminCredentials and replace it with addFireCloudCredentials.

**Comment by davidangb** [comment]
Could you also modify the mock server in UserServiceSpec to check for the presence of the shared-secret header (and its mock value "123") on all relevant endpoints?

done with first-round review!

**Comment by rushtong** [comment]
Good catch on `UserService.remoteGetAllURL`, I'm pretty sure the UI isn't calling that endpoint. It does need to be handled since the user's ID isn't enough - we also need to add the right header. As these cases expand, I'm less and less happy with handling them in individual cases and wonder if a better approach is to add it to all requests regardless of destination.

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala (hunk 1 above)
Comment (rushtong) [reply src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala]: That pipeline is re-used to make a call to Thurloe somewhere below. The profile page definitely breaks without the credentials here.

Comment (rushtong) [reply src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala]: I'm wrong about this - it is not needed in `OAuthService`.

**Comment by rushtong** [comment]
@davidangb - Ready for re-review. Travis isn't happy for something unrelated so I'm looking into why.

**Comment by davidangb** [comment]
Code looks 👍 ... I haven't tested beyond unit tests.

**Pull request merged**